internal/jsre/deps/web3.js

```diff
@@ -5902,35 +6065,28 @@ var eth = function () {
 
 /// @returns an array of objects describing web3.shh.watch api methods
 var shh = function () {
-    var newFilter = new Method({
-        name: 'newFilter',
-        call: 'shh_newFilter',
-        params: 1
-    });
-
-    var uninstallFilter = new Method({
-        name: 'uninstallFilter',
-        call: 'shh_uninstallFilter',
-        params: 1
-    });
-
-    var getLogs = new Method({
-        name: 'getLogs',
-        call: 'shh_getMessages',
-        params: 1
-    });
-
-    var poll = new Method({
-        name: 'poll',
-        call: 'shh_getFilterChanges',
-        params: 1
-    });
 
     return [
-        newFilter,
-        uninstallFilter,
-        getLogs,
-        poll
+        new Method({
+            name: 'newFilter',
+            call: 'shh_newMessageFilter',
+            params: 1
+        }),
+        new Method({
+            name: 'uninstallFilter',
+            call: 'shh_deleteMessageFilter',
+            params: 1
+        }),
+        new Method({
+            name: 'getLogs',
```

Comment:
Not sure about `getLogs`, its name is not something that I would expect to retrieve messages.

For `eth` we don't offer these raw filter methods. Web3 has its own [filter](https://github.com/ethereum/wiki/wiki/JavaScript-API#web3ethfilter) mechanism that calls these methods under the hood. I tried your branch with the updated web3 and I looks like this is not implemented for whisper, @frozeman?

Reply:
According to @frozeman on gitter:
> Yes it’s only used internal and the name is to have one in both namespaces. But as this is the legacy web3.js I don’t want to change it

These are actually not the functions that are available to the user, they are defined in `watches.js` (not `shh.js`), they are used by the filter code and get overwritten when `Shh` gets instantiated. So it's perfectly safe to leave them here, and actually important according to Fabian.